standardization/src/test/scala/za/co/absa/enceladus/standardization/interpreter/stages/TypeParserSuite.scala

```diff
@@ -16,28 +16,391 @@
+      case (DoubleType, TimestampType, _, _)           |
+           (FloatType, TimestampType, _, _)            => s"to_timestamp(CAST(CAST(`%s` AS DECIMAL(${ds.precision},${ds.scale})) AS STRING), '$pattern')"
+      case (DoubleType, DateType, _, _)                |
+           (FloatType, DateType, _, _)                 => s"to_date(CAST(CAST(`%s` AS DECIMAL(${ds.precision},${ds.scale})) AS STRING), '$pattern')"
+      case (StringType, TimestampType, _, _)           =>  s"to_timestamp(`%s`, '$pattern')"
+      case (StringType, DateType, _, _)                => s"to_date(`%s`, '$pattern')"
+      case (TimestampType, DateType, _, _)             => "to_date(`%s`)"
+      case (DateType, TimestampType, _, _)             => "to_timestamp(`%s`)"
+      case (_, TimestampType, _, _)                    => s"to_timestamp(CAST(`%s` AS STRING), '$pattern')"
+      case (_, DateType, _, _)                         => s"to_date(CAST(`%s` AS STRING), '$pattern')"
+      case _                                           => s"CAST(%s AS ${toType.sql})"
+    }
+  }
+
+  private def checkOutput(output: ParseOutput,
+                          path: String,
+                          target: StructField,
+                          baseType: DataType,
+                          pattern: String = "",
+                          decimalSize: DS = DS(0, 0),
+                          timezone: Option[String] = None,
+                          src: String = "sourceField",
+                          testData: TestData = Map.empty
+                         ): Unit = {
+    def doAssert(expected: String, actual: String): Unit = {
+      if (actual!=expected) {
+        println(s"${baseType.typeName}->${target.dataType.typeName} (Pattern: $pattern, Tz: $timezone)")
+        println(s"Exp: $expected")
+        println(s"Act: $actual")
+        assert(actual == expected)
+      }
+    }
+
+    val srcF = fullName(path, src)
+    val boolS = (!target.nullable).toString
+
+    val default = Defaults.getDefaultValue(target) match {
+      case d: Date => s"DATE '${d.toString}'"
+      case t: Timestamp => s"TIMESTAMP('${t.toString}')"
+      case s: String => s
+      case x => x.toString
+    }
+
+    val castString = createCastTemplate(baseType, target.dataType, pattern, decimalSize, timezone).format(srcF)
+
+    val std = s"CASE WHEN ((size(CASE WHEN (($srcF IS NULL) AND $boolS) THEN array(stdNullErr($srcF)) ELSE CASE WHEN (($srcF IS NOT NULL) AND ($castString IS NULL)) THEN array(stdCastErr($srcF, CAST($srcF AS STRING))) ELSE [] END END) = 0) AND ($srcF IS NOT NULL)) THEN $castString ELSE CASE WHEN (size(CASE WHEN (($srcF IS NULL) AND $boolS) THEN array(stdNullErr($srcF)) ELSE CASE WHEN (($srcF IS NOT NULL) AND ($castString IS NULL)) THEN array(stdCastErr($srcF, CAST($srcF AS STRING))) ELSE [] END END) = 0) THEN NULL ELSE $default END END AS `${target.name}`"
+    val actual = output.stdCol.toString().replaceFirst("#\\d+$", "")
```

Comment:
```suggestion
    val actual = output.stdCol.toString()
```
This seems to be unnecessary, as just below the same operation does not require it (`val errActual = output.errors.toString()`).